components/ToReservePageButton/ToReservePageButton.js

```diff
@@ -0,0 +1,31 @@
+import styled from "styled-components";
+import { useRouter } from "next/router";
+
+export default function ToReservePageButton({ id }) {
+  const router = useRouter();
+  function handleToReservePage() {
+    router.push(`/restaurants/reserve/${id}`);
+  }
+  return (
+    <StyledButton
+      type="button"
+      aria-label="reserve"
+      onClick={handleToReservePage}
```

Comment:
Since you are only doing one thing inside the `handle` function, I would recommend to call `router.push` directly inside the `onClick` prop.
```suggestion
      onClick={() => router.push(`/restaurants/reserve/${id}`)}
```